server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/DoubleValuesComparatorSource.java

```diff
@@ -104,7 +104,7 @@ public FieldComparator<?> newComparator(String fieldname, int numHits, boolean e
         final double dMissingValue = (Double) missingObject(missingValue, reversed);
         // NOTE: it's important to pass null as a missing value in the constructor so that
```

Comment:
@gashutos I think the comment does not match the code anymore but it was here for a reason

Reply:
Yeah, this looke very old comments. This was part of this commit https://github.com/opensearch-project/OpenSearch/commit/629f91ae57b5a2223f2edaa213b5d0d08a155885. which is in 2018 but the skipping logic got introduced later.
For skipping logic, if missing values are competitive, we dont invoke skipping optimization as it skipps missing value docuemts through BKD which were not indexed in BKD at all. https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/comparators/NumericComparator.java#L192C1-L193C1

I added integ tests as well to address this scenario.

I am thinking to remove this comment.